src/mobu/services/business/notebookrunner.py

```diff
@@ -53,7 +55,7 @@ class NotebookRunner(NubladoBusiness):
 
     def __init__(
         self,
-        options: NotebookRunnerOptions,
+        options: NotebookRunnerOptions | ListNotebookRunnerOptions,
```

Comment:
We have previously had a one-to-one correspondence between options and instances of a business, and while this is fairly simple, I kind of want to keep that pattern for conceptual simplicity for mobu overall. Would it make sense to have a base Nublado business and then have subclasses that implement the algorithm for determining which notebooks to run? I think that would make it easier to understand the overall structure.

This could be follow-on work in a separate PR, since it's going to be a bit of a rework.

Reply:
The only reason I split these out was because it only makes sense to specify one of `notebooks_to_run` and `max_executions` at the same time.
What about just going back to a single Options, and using a model validator to validate that only one of those options is specified?

Reply:
I dunno, I think having them separate is good! I have separated config and business for similarly trivial things before (like the two TAP runners). I think the two child classes may have only one or two very short methods. But maybe I'm more comfortable with large inheritance hierarchies than I should be.

I don't have any strong feelings about this, to be clear; if you think it's better the way it is, I'm fine with leaving it alone. It just jumped out at me since it felt inconsistent with the pattern for other businesses.

Reply:
The base-business-with-subclasses pattern does seem to fit better.  I don't have super strong feelings either, glad to refactor it later, but yeah, I'll leave it as-is now.